src/main/java/org/fenixedu/academic/ui/struts/action/accounts/ManageAccountsDA.java

```diff
@@ -74,21 +74,17 @@ public ActionForward prepareCreatePerson(ActionMapping mapping, ActionForm form,
     public ActionForward showExistentPersonsWithSameMandatoryDetails(ActionMapping mapping, ActionForm actionForm,
             HttpServletRequest request, HttpServletResponse response) throws Exception {
         PersonBean bean = getRenderedObject("personBean");
-        Collection<Person> results;
-        if (!Strings.isNullOrEmpty(bean.getGivenNames()) || !Strings.isNullOrEmpty(bean.getFamilyNames())) {
+        Collection<Person> results = Collections.emptySet();
+        if (!Strings.isNullOrEmpty(bean.getDocumentIdNumber())) {
+            results = Person.findPersonByDocumentID(bean.getDocumentIdNumber());
+        }
+        if (results.isEmpty() && (!Strings.isNullOrEmpty(bean.getGivenNames()) || !Strings.isNullOrEmpty(bean.getFamilyNames()))) {
             String name =
                     Stream.of(bean.getGivenNames(), bean.getFamilyNames()).filter(n -> !Strings.isNullOrEmpty(n))
                             .collect(Collectors.joining(" "));
             Stream<Person> stream = Person.findPersonStream(name, Integer.MAX_VALUE);
-            if (!Strings.isNullOrEmpty(bean.getDocumentIdNumber())) {
-                stream = stream.filter(p -> p.getDocumentIdNumber().equals(bean.getDocumentIdNumber()));
-            }
             results = stream.collect(Collectors.toSet());
-        } else if (!Strings.isNullOrEmpty(bean.getDocumentIdNumber())) {
-            results = Person.findPersonByDocumentID(bean.getDocumentIdNumber());
-        } else {
-            results = Collections.emptySet();
-        }
+        } 
```

Comment:
The ``results`` collection should keep all the results (searching for ``documentIdNumber`` and ``names``)
With this code it will only search using names if there is no match for the ``documentIdNumber`` which is limitative.